src/rules/no-reversed-media-syntax.js

```diff
@@ -78,61 +72,27 @@ export default {
 		 */
 		function findReversedMediaSyntax(node) {
 			const text = sourceCode.getText(node);
+
+			/** @type {RegExpExecArray} */
 			let match;
 
 			while ((match = reversedPattern.exec(text)) !== null) {
-				const [reversedSyntax, label, url] = match;
-				const matchIndex = match.index;
-				const matchLength = reversedSyntax.length;
-
-				if (
-					isInSkipRange(
-						matchIndex + node.position.start.offset,
-						skipRanges,
-					)
-				) {
+				const { label, url } = match.groups;
+				const [startOffset, endOffset] = match.indices[0].map(
+					index => index + node.position.start.offset,
+				); // Adjust `reversedPattern` match indices to the full source code.
```

Comment:
```suggestion

				// Adjust match index to the full source code.
				const startOffset = match.index + node.position.start.offset;
				const endOffset = startOffset + match[0].length;
```

I think it would be surprising to use the `d` flag and than not use indices of captured substrings. Can we still use the "old" way here? The [RegExp Match Indices proposal](https://github.com/tc39/proposal-regexp-match-indices) also has a note that "For performance reasons, `indices` will only be added to the result if the `d` flag is specified", so for performance reasons it might be better not to use the `d` flag where it isn't needed.